pkg/lang/javascript/plugin_persist.go

```diff
@@ -238,17 +239,25 @@ func (p *persister) transformSecret(file *core.SourceFile, cap *core.Annotation,
 	return result, nil
 }
 
-func (p *persister) transformFS(file *core.SourceFile, cap *core.Annotation, fsR *persistResult) (core.CloudResource, error) {
-	if err := file.ReplaceNodeContent(fsR.expression, "fsRuntime.fs"); err != nil {
-		return nil, errors.Wrap(err, "could not reparse FS transformation")
+func (p *persister) transformFS(unit *core.ExecutionUnit, file *core.SourceFile, cap *core.Annotation, fsR *persistResult) (core.CloudResource, string, error) {
+	if err := file.ReplaceNodeContent(fsR.expression, fmt.Sprintf("fs_%sRuntime.fs", cap.Capability.ID)); err != nil {
+		return nil, "", errors.Wrap(err, "could not reparse FS transformation")
 	}
 
+	fsEnvVar := core.EnvironmentVariable{
+		Name:       cap.Capability.ID + "_bucket_name",
```

Comment:
Ditto on sanitization here.

Reply:
the name is just the env var name, so it doesnt go through sanitization. The actual bucket name will be the value and that is dynamically generated based on the sanitized bucket name

Reply:
Do annotation ids have any restriction, do you know offhand? For example, can they contain spaces? (afaik, an env var can't.)

Reply:
yeah the annotation ids are the same regex as the app name, so i believe that it should be valid env vars.